### PR TITLE
Improve and generate documentation

### LINKED ISSUE #18: Generate API documentation
### What's needed?

Right now users only can only find out about what's available in this project by digging into the code.

### Proposed solution

Generate API documentation either from the protocol buffers files or the python files (or both).

### Use cases

_No response_

### Alternatives and workarounds

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leandro-lucarella-frequenz**
- Generate documentation for proto files
- Improve documentation and homogenize units

### CODE REVIEW

File: proto/frequenz/api/microgrid/common.proto [hunk 1]
```diff
@@ -145,91 +148,94 @@ message AC {
...
     // consumption and supply.
-    // In Watthours (Wh).
+    // In Watt-hour (Wh).
     ActiveEnergy energy_active = 6;
 
     // The total reactive energy counters for the underlying component's
     // capacitive and inductive energy values.
-    // In VArh.
+    // In Volt-Ampere reactive hour (VArh).
+    // FIXME: ReactiveEnergy says it's in VAh
```
Comment (leandro-lucarella-frequenz) [inline proto/frequenz/api/microgrid/common.proto]: Need feedback here.

File: proto/frequenz/api/microgrid/common.proto [hunk 2]
```diff
@@ -145,91 +148,94 @@ message AC {
...
     ReactiveEnergy energy_reactive = 7;
 
     // Harmonics of the instantaneous active power at the component.
     Harmonics harmonics_power_active = 8;
 
     // Total harmonic distortion
     // of the instantaneous active power at the component.
+    // FIXME: Unit?
```
Comment (leandro-lucarella-frequenz) [inline proto/frequenz/api/microgrid/common.proto]: Need feedback here.

File: proto/frequenz/api/microgrid/common.proto [hunk 3]
```diff
@@ -145,91 +148,94 @@ message AC {
...
   // consumption and supply.
-  // In Watthours (Wh).
+  // In Watt-hour (Wh).
   ActiveEnergy energy_active = 7;
 
   // The total 3-phase reactive energy counters for the underlying component's
   // capacitive and inductive energy values.
-  // In VArh.
+  // In Volt-Ampere reactive hour (VArh).
+  // FIXME: ReactiveEnergy says Volt-Ampere-hour (VAh).
```
Comment (leandro-lucarella-frequenz) [inline proto/frequenz/api/microgrid/common.proto:239]: Need feedback here.

**Review by leandro-lucarella-frequenz — COMMENTED** [review]
@tiyash-basu-frequenz I need some feedback on a few places where I'm not sure if the docs are correct or complete.

**Comment by leandro-lucarella-frequenz** [comment]
Generated docs example: https://github.com/frequenz-floss/frequenz-api-microgrid/actions/runs/3196021026/attempts/1#summary-8739594315

I didn't find a way to link directly to the latest run, and not as useful as it could be, as links don't work in there, but still, is a first step.

FYI too @jakub-toptal

File: proto/frequenz/api/microgrid/common.proto [hunk 4]
```diff
@@ -145,91 +148,94 @@ message AC {
...
     // -ve power means supply into the grid.
-    // In Watts (W).
+    // In Watt (W).
     Metric power_active = 3;
 
     // AC reactive power.
     // +ve power means inductive (leading).
     // -ve power means capacitive (lagging).
-    // In VArh.
+    // In Volt-Ampere reactive hour (VArh).
```
Comment (tiyash-basu-frequenz) [inline proto/frequenz/api/microgrid/common.proto]: The old doc was incorrect. It should be just `Volt-Ampere reactive`.

In general, power units do not have a time component in them, e.g. it's `Watt`, and not `Watt-hour`.

**Comment by leandro-lucarella-frequenz** [comment]
Part of:
* #18

File: proto/frequenz/api/microgrid/common.proto [hunk 5]
```diff
@@ -73,19 +74,21 @@ message MetricAggregation {
 // Metrics of a DC electrical connection.
 message DC {
   // The DC voltage across the component.
-  // In Volts (V).
+  // In Volt (V).
```
Comment (tiyash-basu-frequenz) [inline proto/frequenz/api/microgrid/common.proto:77]: I understand that this is being done for consistency, but in text or in speech, plural units are more natural for some metrics. E.g., we say 230 volts and 5 amperes, and not 220 volt and 5 ampere.

File: proto/frequenz/api/microgrid/common.proto (hunk 1 above)
Comment (tiyash-basu-frequenz) [reply proto/frequenz/api/microgrid/common.proto]: Reactive energy is in `VArh`.

Here are some units for context: 
Active power: Watts (W)
Active energy: Watt-hours (Wh) (Note the time component)
Reactive power: Volt-ampere-reactive (VAr) (Note the 'r' for reactive)
Reactive energy: Volt-ampere-reactive-hours (VArh)
Apparent power: Volt-amperes (VA) (Note the 'r' for reactive)
Apparent energy: Volt-ampere hours (VAh)

File: proto/frequenz/api/microgrid/common.proto (hunk 2 above)
Comment (tiyash-basu-frequenz) [reply proto/frequenz/api/microgrid/common.proto]: It's in percent (%).

Harmonics are also in percent (%).

File: proto/frequenz/api/microgrid/common.proto (hunk 3 above)
Comment (tiyash-basu-frequenz) [reply proto/frequenz/api/microgrid/common.proto:239]: Reactive energy is in `VArh`.

File: proto/frequenz/api/microgrid/common.proto [hunk 6]
```diff
@@ -11,6 +11,7 @@ syntax = "proto3";
 package frequenz.api.microgrid.common;
 
 // A set of lower and upper bounds for any metric.
+// Bounds units are always the same as the related metric.
```
Comment (tiyash-basu-frequenz) [inline proto/frequenz/api/microgrid/common.proto]: I'd rephrase this slightly:
`Bounds units` => `Units of the bounds`

File: proto/frequenz/api/microgrid/common.proto (hunk 1 above)
Comment (leandro-lucarella-frequenz) [reply proto/frequenz/api/microgrid/common.proto]: What I mean is in the definition of the `ReactiveEnergy` message, it says is in VAh:

https://github.com/frequenz-floss/frequenz-api-microgrid/blob/0ffff8010646d9d824367021be9a5bc99aee7b5a/proto/frequenz/api/microgrid/common.proto#L112-L127

File: proto/frequenz/api/microgrid/common.proto (hunk 6 above)
Comment (leandro-lucarella-frequenz) [reply proto/frequenz/api/microgrid/common.proto]: Done

File: proto/frequenz/api/microgrid/common.proto (hunk 5 above)
Comment (leandro-lucarella-frequenz) [reply proto/frequenz/api/microgrid/common.proto:77]: It is for consistency and because it's specifying the unit (Volt, Ampere, Watt, etc.), is not part of a phrase speaking about a quantity, like 220 Volts. For me it sounds more correct to specify units in singular, but if you think it's clearer to specify units in plural, I can change it, no problem.

File: proto/frequenz/api/microgrid/common.proto (hunk 4 above)
Comment (leandro-lucarella-frequenz) [reply proto/frequenz/api/microgrid/common.proto]: Fixed

File: proto/frequenz/api/microgrid/common.proto (hunk 1 above)
Comment (leandro-lucarella-frequenz) [reply proto/frequenz/api/microgrid/common.proto]: I updated `ReactiveEnergy` to use `VArh`, I guess that should be correct.

**Comment by leandro-lucarella-frequenz** [comment]
Updated to address all the comments. I just noticed `ActiveEnergy` and `ReactiveEnergy` has as comments:

> // The amount of {active,reactive} power the inverter is consuming or generating.

Is this correct? Shouldn't it be somehow {active,reactive} **energy** instead of power?

File: proto/frequenz/api/microgrid/common.proto (hunk 1 above)
Comment (tiyash-basu-frequenz) [reply proto/frequenz/api/microgrid/common.proto]: Oh, right. `VArh` is correct.

File: proto/frequenz/api/microgrid/common.proto (hunk 5 above)
Comment (tiyash-basu-frequenz) [reply proto/frequenz/api/microgrid/common.proto:77]: No need to change it :) This is alright.

**Comment by tiyash-basu-frequenz** [comment]
> Updated to address all the comments. I just noticed `ActiveEnergy` and `ReactiveEnergy` has as comments:
> 
> > // The amount of {active,reactive} power the inverter is consuming or generating.
> 
> Is this correct? Shouldn't it be somehow {active,reactive} **energy** instead of power?

Right, it should be energy.

**Comment by leandro-lucarella-frequenz** [comment]
> Right, it should be energy.

Alright, updated again!

**Review by tiyash-basu-frequenz — APPROVED** [review]
Thanks! LGTM!
